Layout.zone_primitives failing

### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

When trying to retrieve zone primitives (the layout is multizone) we got this error.


![image](https://github.com/user-attachments/assets/c4c60790-05e4-4a1d-bf82-880e759fd036)

![image](https://github.com/user-attachments/assets/77a32d62-0e31-42ab-ae2f-dc9dc4317c74)

![image](https://github.com/user-attachments/assets/eb822032-d792-4884-ac25-626592a91f4c)


self = <pyedb.grpc.edb_core.layout.layout.Layout object at 0x000001B9D2BEAA90>

    @property
    def zone_primitives(self):
        """:obj:`list` of :class:`.Primitive`: List of \
        all primitives in the :term:`zones <Zone>`.
    
        This property is read-only.
        """
>       return [Primitive(msg) for msg in self.__stub.GetZonePrimitives(self.msg)]
E       TypeError: 'EDBObjCollectionMessage' object is not iterable

### 📝 Steps to reproduce

NA

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

_No response_

### 🐍 Which Python version are you using?

3.10

### 📦 Installed packages

```shell
NA
```

--- a/.github/workflows/ci_cd.yml
+++ b/.github/workflows/ci_cd.yml
@@ -33,9 +33,10 @@ jobs:
       runs-on: ubuntu-latest
       steps:
         - name: PyAnsys documentation style checks
-          uses: ansys/actions/doc-style@v5
+          uses: ansys/actions/doc-style@v8
           with:
             token: ${{ secrets.GITHUB_TOKEN }}
+            vale-version: "2.29.6"
 
   smoke-tests:
     name: Build and Smoke tests

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,7 @@ classifiers = [
 
 # FIXME: add ansys-api-edb version
 dependencies = [
-    "ansys-api-edb==1.0.9",
+    "ansys-api-edb==1.0.10",
     "protobuf>=3.19.3,<5",
     "grpcio>=1.44.0"
 ]

--- a/src/ansys/edb/core/geometry/polygon_data.py
+++ b/src/ansys/edb/core/geometry/polygon_data.py
@@ -302,7 +302,7 @@ class PolygonData:
         PolygonData
         """
         return self.__stub.Transform(
-            messages.polygon_data_transform_message("scale", factor, center)
+            messages.polygon_data_transform_message("scale", self, factor, center)
         )
 
     @parser.to_polygon_data

--- a/src/ansys/edb/core/hierarchy/component_group.py
+++ b/src/ansys/edb/core/hierarchy/component_group.py
@@ -29,7 +29,7 @@ class ComponentGroup(Group):
     __stub: ComponentGroupServiceStub = StubAccessor(StubType.component_group)
 
     @classmethod
-    def create_with_component(cls, layout, name, comp_name):
+    def create(cls, layout, name, comp_name):
         """Create a component group with a component.
 
         Parameters

--- a/src/ansys/edb/core/inner/messages.py
+++ b/src/ansys/edb/core/inner/messages.py
@@ -260,9 +260,9 @@ def polygon_data_pair_with_tolerance_message(pd1, pd2, tol):
     )
 
 
-def _polygon_data_transform_message_point_value(point, value):
+def _polygon_data_transform_message_point_value(val, point):
     """Convert to a ``PolygonDataTransformMessage`` object."""
-    return PolygonDataTransformMessage.PointValueMessage(point=point_message(point), value=value)
+    return PolygonDataTransformMessage.PointValueMessage(point=point_message(point), value=val)
 
 
 def polygon_data_transform_message(op, pd, *args):
@@ -481,7 +481,7 @@ def transform_message(transform):
     else:
         return TransformMessage(
             scale=value_message(transform.scale),
-            angle=value_message(transform.angle),
+            angle=value_message(transform.rotation),
             mirror=transform.mirror,
             offset_x=value_message(transform.offset_x),
             offset_y=value_message(transform.offset_y),

--- a/src/ansys/edb/core/layout/layout.py
+++ b/src/ansys/edb/core/layout/layout.py
@@ -273,7 +273,7 @@ class Layout(ObjBase, variable_server.VariableServer):
 
         This property is read-only.
         """
-        return [Primitive(msg) for msg in self.__stub.GetZonePrimitives(self.msg)]
+        return [Primitive(msg).cast() for msg in self.__stub.GetZonePrimitives(self.msg).items]
 
     @property
     def fixed_zone_primitive(self):

--- a/src/ansys/edb/core/primitive/primitive.py
+++ b/src/ansys/edb/core/primitive/primitive.py
@@ -1220,22 +1220,24 @@ class Bondwire(Primitive):
         Parameters
         ----------
         x1 : :class:`.Value`
-            X value of the the start point.
+            X value of the start point.
         y1 : :class:`.Value`
-            Y value of the the start point.
+            Y value of the start point.
         x2 : :class:`.Value`
-            X value of the the end point.
+            X value of the end point.
         y2 : :class:`.Value`
             Y value of the end point.
         """
         self.__stub.SetTraj(
-            target=self.msg,
-            traj=bondwire_pb2.BondwireTrajMessage(
-                x1=messages.value_message(x1),
-                y1=messages.value_message(y1),
-                x2=messages.value_message(x2),
-                y2=messages.value_message(y2),
-            ),
+            bondwire_pb2.SetBondwireTrajMessage(
+                target=self.msg,
+                traj=bondwire_pb2.BondwireTrajMessage(
+                    x1=messages.value_message(x1),
+                    y1=messages.value_message(y1),
+                    x2=messages.value_message(x2),
+                    y2=messages.value_message(y2),
+                ),
+            )
         )
 
     @property
@@ -1512,7 +1514,7 @@ class PadstackInstance(conn_obj.ConnObj):
             padstack_instance_pb2.PadstackInstSetSolderBallLayerMessage(
                 target=self.msg,
                 layer=solderball_layer.msg,
-            )(self, solderball_layer)
+            )
         )
 
     @property

--- a/src/ansys/edb/core/simulation_setup/simulation_settings.py
+++ b/src/ansys/edb/core/simulation_setup/simulation_settings.py
@@ -227,6 +227,15 @@ class AdvancedSettings(SimulationSettingsBase):
         self.__stub.SetViaMaterial(messages.string_property_message(self, via_material))
 
     @property
+    def mesh_for_via_plating(self):
+        """:obj:`bool`: Flag indicating if meshing for via plating is enabled."""
+        return self.__stub.GetMeshForViaPlating(self.msg).value
+
+    @mesh_for_via_plating.setter
+    def mesh_for_via_plating(self, mesh_for_via_plating):
+        self.__stub.SetMeshForViaPlating(messages.bool_property_message(self, mesh_for_via_plating))
+
+    @property
     def model_type(self):
         """:class:`.ModelType`: model type."""
         return ModelType(self.__stub.GetModelType(self.msg).defeature_model_type)

--- a/tests/e2e/scratch/sim_setup_scratch.py
+++ b/tests/e2e/scratch/sim_setup_scratch.py
@@ -141,6 +141,10 @@ def do_advanced_test(settings: HFSSAdvancedSettings):
     settings.via_material = "new_via_material"
     new_via_material = settings.via_material
 
+    og_mesh_via_plating = settings.mesh_for_via_plating
+    settings.mesh_for_via_plating = not og_mesh_via_plating
+    new_mesh_via_plating = settings.mesh_for_via_plating
+
     og_ic_mode_auto_resolution = settings.ic_mode_auto_resolution
     settings.ic_mode_auto_resolution = not og_ic_mode_auto_resolution
     new_ic_mode_auto_resolution = settings.ic_mode_auto_resolution

